Source/JavaScriptCore/jsc.cpp

```diff
@@ -1389,12 +1389,25 @@ JSC_DEFINE_HOST_FUNCTION(functionAtob, (JSGlobalObject* globalObject, CallFrame*
 {
     VM& vm = globalObject->vm();
     auto scope = DECLARE_THROW_SCOPE(vm);
-    auto* string = callFrame->argument(0).toString(globalObject);
+
+    if (!callFrame->argumentCount())
+        return JSValue::encode(throwException(globalObject, scope, createError(globalObject, "Missing input for atob."_s)));
+
+    JSValue jsValue = callFrame->argument(0);
+    if (jsValue.isUndefined())
+        return JSValue::encode(throwException(globalObject, scope, createError(globalObject, "Invalid argument for atob."_s)));
```

Comment:
Is this the expected behavior of the Web API?  Ditto for all the checks below?

Reply:
Yes, I believe so. Web API Spec: https://html.spec.whatwg.org/multipage/webappapis.html#dom-atob-dev